.github/workflows/dependabotautomerge.yml

```diff
@@ -0,0 +1,30 @@
+# Copyright 2021 Security Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: auto-merge
+
+on:
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  auto-merge:
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
Can we add an additional step here that validate PRs are from dependabot? This would reduce the attack surface.

Reply:
the action seems to be doing that in https://github.com/ahmadnassri/action-dependabot-auto-merge/blob/master/action/index.js#L21
What I don't understand is how the `login` is set: it does not seem to correspond to an actual GitHub user (`github.com/dependabot[bot]` or even `github.com/dependabot-preview` do not exist). So there seems to be a way to set it. Is this a feature only accessible to GitHub or to arbitrary users?

Maybe I understand this all wrong, please feel free to correct me